manifests/exception.pp

```diff
@@ -133,16 +136,11 @@
     }
 
     # Validate common parameters
-    validate_re($ensure,['^(present|absent)$'])
     validate_slength($display_name,255)
-    validate_re($enabled,['^(yes|no)$'])
-    validate_re($allow_edge_traversal,['^(yes|no)$'])
 
     case $::operatingsystemversion {
       'Windows Server 2012', 'Windows Server 2008', 'Windows Server 2008 R2', 'Windows Vista','Windows 7','Windows 8': {
         validate_slength($description,255)
```

Comment:
Can be ```String[0, 255]```

Reply:
@ekohl I don't think we can do this with data type limit.
I'm pretty sure this limit holds only true for those OSes mentioned - else, why would it be coded this way!?

Reply:
You can switch to assert_type though https://puppet.com/docs/puppet/5.3/function.html#asserttype